modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/DataStructuresProcessor.java

```diff
@@ -1050,7 +1061,7 @@ public void removeQueue(final String name, final GridCacheContext cctx) throws I
         final IgniteInternalCache cache;
 
         if (create) {
-            cache = compatibleCache(cfg, grpName);
+            cache = compatibleCache(cfg, grpName, !(type == DataStructureType.SET && !cfg.isCollocated()), name);
```

Comment:
May be replace
!(type == DataStructureType.SET && !cfg.isCollocated())
to
type != DataStructureType.SET || cfg.isCollocated()
?